cylc/flow/task_queue.py

```diff
@@ -0,0 +1,201 @@
+            if qname == "type":
+                continue
+            if qname == self.Q_DEFAULT:
+                # Add all tasks to the default queue.
+                qmembers = set(all_task_names)
+            else:
+                qmembers = set()
+                for mem in queue["members"]:
+                    if mem in descendants:
+                        # Family name.
+                        for fmem in descendants[mem]:
+                            if (fmem not in descendants
+                                    and fmem in all_task_names):
+                                # Task name.
+                                qmembers.add(fmem)
+                    elif mem in all_task_names:
+                        # Task name.
+                        qmembers.add(mem)
+            queues[qname] = {}
+            queues[qname]["members"] = qmembers
+            queues[qname]["limit"] = queue["limit"]
+        return queues
+
+    def _make_indep(self, in_queues: dict) -> dict:
+        """For classic queues tasks cannot belong to multiple limit groups.
+
+        Latest assigned group takes precedence. So, for example, the "default"
+        group will only contain tasks not assigned to any other queue.
+
+        """
+        queues: Dict[str, Any] = {}
+        seen: Dict[str, str] = {}
+        for qname, qconfig in in_queues.items():
+            queues[qname] = {}
+            queues[qname]["members"] = qconfig["members"]
+            queues[qname]["limit"] = qconfig["limit"]
+            if qname == self.Q_DEFAULT:
+                continue
+            for qmem in qconfig["members"]:
+                # Remove from default queue
+                try:
+                    queues[self.Q_DEFAULT]["members"].remove(qmem)
+                except KeyError:
+                    # Already removed.
+                    pass
+                if qmem in seen:
+                    # Override previous queue assignment.
+                    oldq = seen[qmem]
+                    queues[oldq]["members"].remove(qmem)
+                else:
+                    queues[qname]["members"].add(qmem)
+                seen[qmem] = qname
+        return queues
+
+    def _configure_limiters(self, queues: dict) -> None:
+        """Create a Limiter for each limit group."""
+        for name, config in queues.items():
+            self.limiters[name] = Limiter(config["limit"], config["members"])
+
+    def add(self, itask: TaskProxy) -> None:
+        """Queue a task."""
+        LOG.debug(f"Queue add: {itask.identity}")
+        itask.state.reset(is_queued=True)
+        itask.reset_manual_trigger()
+        self.task_deque.appendleft(itask)
+
+    def _is_limited(self, itask: TaskProxy, active: Counter[str]) -> bool:
+        """Return True if the task is limited, else False."""
+        for name, limiter in self.limiters.items():
+            if limiter.is_limited(itask, active):
+                return True
+        return False
```

Comment:
I think it could be too:
```
return any(
  limiter.is_limited(itask, active)
  for name, limiter in self.limiters.items()
)
```
Same but a bit easier to read?

Reply:
Good idea. In fact, single statement then so I'll get rid of the function.